src/App.js

```diff
@@ -1,38 +1,87 @@
 import React, { Component } from 'react';
 import './App.css';
 
+const BlockType = {
+  Heading: 'heading',
+  Text: 'text',
+  Image: 'image',
+  Blockquote: 'blockquote',
+};
+
 class App extends Component {
   state = {
     blocks: [
-      { type: 'heading', text: 'hello world' },
-      { type: 'text', text: 'another string' },
-      { type: 'image', url: 'http://fpoimg.com/300x300', text: 'FPO image' },
-      { type: 'blockQuote', text: 'this is a quote' },
-    ].map(this.withKeys),
+      { type: BlockType.Heading, text: 'hello world' },
+      { type: BlockType.Text, text: 'another string' },
+      {
+        type: BlockType.Image,
+        url: 'http://fpoimg.com/300x300',
+        text: 'FPO image',
+      },
+      { type: BlockType.Blockquote, text: 'this is a quote' },
+    ].map(this.toObjectsWithKeys),
   };
 
-  withKeys(block, index) {
+  toObjectsWithKeys(block, index) {
     const key = btoa(`${index}: ${JSON.stringify(block)}`);
     return { key, ...block };
   }
 
+  // step 2: refactor the addtype functions into generic add block function where you just pass in the block.type as an argument
+  addType = option => {
+    option = BlockType;
+    const newOption = this.toObjectsWithKeys(
+      { type: BlockType.Heading, text: 'testing' },
+      this.state.blocks.length,
+    );
+    this.setState({ blocks: [...this.state.blocks, newOption] });
+  };
+
   addHeading = () => {
-    const newHeading = this.withKeys(
-      { type: 'heading', text: 'testing' },
+    const newHeading = this.toObjectsWithKeys(
+      { type: BlockType.Heading, text: 'testing' },
       this.state.blocks.length,
     );
     this.setState({ blocks: [...this.state.blocks, newHeading] });
   };
 
+  addText = () => {
+    const newText = this.toObjectsWithKeys(
+      { type: BlockType.Text, text: 'Test number 2 on the button' },
+      this.state.blocks.length,
+    );
+    this.setState({ blocks: [...this.state.blocks, newText] });
+  };
+
+  addImage = () => {
+    const newImage = this.toObjectsWithKeys(
+      {
+        type: BlockType.Image,
+        url: 'http://fpoimg.com/300x200',
+        text: 'FPO image',
+      },
+      this.state.blocks.length,
+    );
+    this.setState({ blocks: [...this.state.blocks, newImage] });
+  };
+
+  addBlockquote = () => {
+    const newBlockquote = this.toObjectsWithKeys(
+      { type: BlockType.Blockquote, text: 'this is a NEW quote' },
+      this.state.blocks.length,
+    );
+    this.setState({ blocks: [...this.state.blocks, newBlockquote] });
+  };
+
   toComponents(block) {
```

Comment:
Same here, lets to `toComponent` and keep these "mappers" singular going forward.